iceoryx_dds/include/iceoryx_dds/internal/gateway/iox_to_dds.inl

```diff
@@ -0,0 +1,118 @@
+// Copyright (c) 2020 by Robert Bosch GmbH. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <chrono>
+#include <thread>
+
+#include <iceoryx_posh/mepoo/chunk_header.hpp>
+
+#include "iceoryx_dds/internal/log/logging.hpp"
+
+namespace iox
+{
+namespace dds
+{
+// template <typename channel_t>
+// using ChannelFactory = std::function<channel_t(const iox::capro::ServiceDescription)>;
+
+// ======================================== Public ======================================== //
+template <typename channel_t, typename gateway_t>
+inline Iceoryx2DDSGateway<channel_t, gateway_t>::Iceoryx2DDSGateway() noexcept : gateway_t()
+{
+}
+
+template <typename channel_t, typename gateway_t>
+inline void Iceoryx2DDSGateway<channel_t, gateway_t>::loadConfiguration(GatewayConfig config) noexcept
+{
+    iox::LogDebug() << "[Iceoryx2DDSGateway] Configuring gateway.";
+    for (const auto& service : config.m_configuredServices)
+    {
+        if (!this->findChannel(service).has_value())
+        {
+            auto channel = this->addChannel(service);
+            auto subscriber = channel.getIceoryxTerminal();
```

Comment:
getIceoryxTerminal might be a shared_ptr which contains a nullptr

Reply:
:heavy_check_mark: